src/sas/sascalc/dataloader/readers/cansas_reader_HDF5.py

```diff
@@ -67,14 +63,24 @@ def read(self, filename):
         """
         # Reinitialize when loading a new data file to reset all class variables
         self.reset_class_variables()
+
+        filename = self.f_open.name
+        self.f_open.close() # IO handled by h5py
+
         # Check that the file exists
         if os.path.isfile(filename):
             basename = os.path.basename(filename)
             _, extension = os.path.splitext(basename)
             # If the file type is not allowed, return empty list
             if extension in self.ext or self.allow_all:
                 # Load the data file
-                self.raw_data = h5py.File(filename, 'r')
+                try:
+                    self.raw_data = h5py.File(filename, 'r')
+                except Exception as e:
+                    if extension not in self.ext:
+                        msg = "CanSAS2.0 HDF5 Reader could not load file {}".format(basename + extension)
+                        raise DefaultReaderException(msg)
+                    raise FileContentsException(e.message)
                 # Read in all child elements of top level SASroot
                 self.read_children(self.raw_data, [])
```

Comment:
Returning self.output below here is no longer needed as the parent class now handles that.